### ISSUE TITLE
[JetBrains] Terminals don't resize properly causing text to overflow

### ISSUE THREAD

**Issue by andreafalzetti** [gitpod-io/gitpod#15385]
### Bug description

When using JetBrains IDE and task terminals (via attach), the resize is not propagated correctly causing the rendering to break.

Example
![2022-12-14 12 08 25](https://user-images.githubusercontent.com/2318450/208074646-dd814725-bd98-409d-81db-1300a624c6ae.gif)

Related YouTrack issue: https://youtrack.jetbrains.com/issue/GTW-2744/Terminal-rendering-breaking-when-typing

### Steps to reproduce

Use task terminals and type a long string

**Comment by felladrin** [comment]
ℹ️  We've received feedback from a user saying they're using `shopt -u checkwinsize` [[1](https://www.gnu.org/software/bash/manual/html_node/The-Shopt-Builtin.html)] to workaround this problem.

**Comment by felladrin** [comment]
ℹ️ We've also confirmed the [terminal resizing](https://youtrack.jetbrains.com/issue/CWM-4314/htop-doesnt-render-properly-in-JetBrains-Client-Terminal) (which could be related to this issue) is working great on IntelliJ 2022.3.2, but this issue with Gitpod Task Terminals reported above, is still present in this version.

**Comment by felladrin** [comment]
I tried resolving this issue today, but nothing I tried has fixed this issue.

But I found a temporary workaround which is right-clicking the Terminal Tab and selecting "Move to Editor":
<img width="298" alt="image" src="https://user-images.githubusercontent.com/418083/212375748-14e88b77-bc01-49cf-880a-5609d7c63e09.png">

By doing this, the terminal stops being attached to the Backend IDE size.

I tried programmatically moving the terminals to the editor area [[1](https://github.com/JetBrains/intellij-community/blob/9e8eefcc93893a0082e910b54d787910b9822a60/plugins/terminal/src/org/jetbrains/plugins/terminal/action/MoveTerminalSessionToEditorAction.kt#L14)], but when I started the IDE, the terminals continued being displayed in the Terminal Widgets area.
